CadbaseLibrary/CadbaseLibrary.FCMacro

```diff
@@ -0,0 +1,837 @@
+        self.label.setText(QtGui.QApplication.translate("CADBaseLibrary", "Warning: You need write permission on this server"))
+        self.groupBox_3.setTitle(QtGui.QApplication.translate("CADBaseLibrary", "Library path"))
+        self.lineEdit_3.setToolTip(QtGui.QApplication.translate("CADBaseLibrary", "Enter the path to your CADBase library"))
+        self.pushButton_3.setToolTip(QtGui.QApplication.translate("CADBaseLibrary", "Browse to your path library"))
+        self.pushButton_3.setText(QtGui.QApplication.translate("CADBaseLibrary", "..."))
+        self.groupBox_4.setTitle(QtGui.QApplication.translate("CADBaseLibrary", "Password"))
+        self.lineEdit_4.setToolTip(QtGui.QApplication.translate("CADBaseLibrary", "Enter yout password from the CADBase"))
+
+    def setdefaulturl(self):
+        self.lineEdit.setText(cdbs_api)
+
+    def changepath(self):
+        librarypath = param.GetString('destination','')
+        np = QtGui.QFileDialog.getExistingDirectory(self,"Local location of your existing CADBase library",librarypath)
+        if np:
+            self.lineEdit_3.setText(np)
+
+    def accept(self):
+        if self.lineEdit.text():
+            param.SetString('api-url',self.lineEdit.text())
+        if self.lineEdit_3.text():
+            param.SetString('destination',self.lineEdit_3.text())
+
+        if self.lineEdit_2.text() and self.lineEdit_4.text():
+            username, password = (self.lineEdit_2.text(), self.lineEdit_4.text())
+            CDBSAuth(username, password)
+
+        # FreeCAD.Console.PrintWarning(param.GetString('api-url',''))
+        # FreeCAD.Console.PrintWarning(param.GetString('auth-token',''))
+
+        QtGui.QDialog.accept(self)
+
+if QtCore.QDir(LIBRARYPATH).exists():
+    m = FreeCADGui.getMainWindow()
+    w = m.findChild(QtGui.QDockWidget,"CADBaseLibrary")
+    if w:
+        if hasattr(w,"isVisible"):
+            if w.isVisible():
+                w.hide()
+            else:
+                w.show()
+        else:
+            # something went wrong with our widget... Recreate it
+            del w
+            m.addDockWidget(QtCore.Qt.RightDockWidgetArea,ExpDockWidget(LIBRARYPATH))
+    else:
+        m.addDockWidget(QtCore.Qt.RightDockWidgetArea,ExpDockWidget(LIBRARYPATH))
+else:
+    print("Library path ", LIBRARYPATH, "not found.")
+
+
+def updateComponentsList(LIBRARYPATH):
+
+    data = parsingGpl(LIBRARYPATH)
+    if isinstance(data, SimpleNamespace):
+
+        if len(data.components) == 0:
+
+            FreeCAD.Console.PrintWarning('You don\'t have favorite components\n')
+
+        else:
+
+            for component in data.components:
+                FreeCAD.Console.PrintWarning(component.uuid + '\n')
+                new_dir = LIBRARYPATH + "/" + component.name + ' (from ' + component.ownerUser.username + ')'
+                # FreeCAD.Console.PrintWarning(new_dir)
+
+                if not os.path.exists(new_dir):
```

Comment:
With `pathlib.Path` this would be `if not new_dir.exists():`, which is nicer.

Reply:
Corrected.